tianshou/highlevel/params/lr_scheduler.py

```diff
@@ -25,6 +25,7 @@ def create_scheduler(self, optim: torch.optim.Optimizer) -> LRScheduler:
 
     class _LRLambda:
         def __init__(self, sampling_config: SamplingConfig):
+            assert sampling_config.step_per_collect is not None
             self.max_update_num = (
                 np.ceil(sampling_config.step_per_epoch / sampling_config.step_per_collect)
                 * sampling_config.num_epochs
```

Comment:
How do we handle this properly for the case where the user specifies `episode_per_collect` instead?

Reply:
It seems to me that we could only handle it properly if the epoch was also specified in terms of episodes (i.e. if there was `episode_per_epoch`).

Reply:
Not sure it can be handled truly properly, episode length often changes over the course of a training, so scheduling based may be a bad option. Do that in a follow-up issue? It's unrelated to current changes

Reply:
Maybe the schedule-free optimizer works so well that we don't need to think too much about it ^^